tools/build-version.go

```diff
@@ -38,20 +38,13 @@ func getTag(match ...string) (string, *semver.PRVersion) {
 }
 
 func main() {
-	tags, err := exec.Command("git", "tag").Output()
-	if err != nil {
-		log.Println("git tag:", err)
+	// Fetch the latest tags
+	if err := exec.Command("git", "fetch", "--tags").Run(); err != nil {
```

Comment:
Yes, the action will do a overwrite of the `nightly` tag.
A easy workaround would be, to do the `git fetch --tags` with `-f` or `--force` respectively.
I realized this happens only in the non-forked (upstream) repository.

Currently I try to get at a string of this error too.

Reply:
...Is it actually a good idea to run `git fetch` in `build-version.go` (*especially* every time, *especially* forced fetch)?
`build-version.go` is run by `make` when building micro, for example. Do we want it to silently modify the repo?

Seems like `build-version.go` should only report the version, not produce any side effects?

Since we have now `fetch-depth: 0` and `fetch-tags` in the action config, can we get rid of `git fetch` here entirely?

Reply:
By the way: https://github.com/zyedidia/micro/issues/1504

Reply:
Good idea, done.